### PR TITLE
Change order after transaction failed with the Error

### PR DESCRIPTION
**Pull request by OrenZak**
#### Main purpose:
Change the order if happened any failure on the client side that the server should know about.
For example, if a `sendTransaction` -> failed so we can notify the server about it and he will stop listening for the payment to be received on the blockchain side. So the user/SDK developer will get a failure faster.
#### Technical description:
- Added `changeOrder` API on `OrderRemoteData.java`
- Added login on `OrderRepository.java` so when payment is failed we call `changeOrder` with Transaction failed `Error`.
- Removed the cancel call on `onTransactionFailed` callback from purchase call, because as @doodyparizada  said we can't cancel an order if it already created.
#### Tests:
- Changed `purchase` test to verify we are calling to `remote.changeOrder` when the payment is failed, and never call it if the payment is succeeded.
#### Documentation:
- Added on `OrderRepository.java` for `changeOrder` method.

### CODE REVIEW

File: core/src/main/java/com/kin/ecosystem/core/data/order/OrderRemoteData.java [hunk 1]
```diff
@@ -243,6 +244,50 @@ public void getFilteredOrderHistory(@Nullable String origin, @NonNull String off
...
+                    executorsUtil.mainThread().execute(new Runnable() {
+                        @Override
+                        public void run() {
+                            callback.onResponse(result);
+                        }
+                    });
+				}
+
+				@Override
+				public void onUploadProgress(long bytesWritten, long contentLength, boolean done) {
```
Comment (yosriz) [inline core/src/main/java/com/kin/ecosystem/core/data/order/OrderRemoteData.java]: why are those callbacks needed? what do we upload here? I understand it's auto generated from the swagger definition, but still wondering

File: core/src/main/java/com/kin/ecosystem/core/network/model/Body.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+ */
+
+
+import com.google.gson.annotations.SerializedName;
+import java.util.Objects;
+
+/**
+ * Body
+ */
+public class Body {
```
Comment (yosriz) [inline core/src/main/java/com/kin/ecosystem/core/network/model/Body.java]: what's the purpose of this `Body` class? it seems like a wrapper for error, we can't we use Error directly?

File: core/src/test/java/com/kin/ecosystem/core/data/order/OrderRepositoryTest.java [hunk 3]
```diff
@@ -548,6 +550,7 @@ public void onFailure(KinEcosystemException exception) {
 
 		verify(blockchainSource).removePaymentObserver(observersList.get(0));
 		verify(blockchainSource).removePaymentObserver(observersList.get(1));
+		verify(remote).changeOrder(anyString(), any(Body.class), any(Callback.class));
```
Comment (yosriz) [inline core/src/test/java/com/kin/ecosystem/core/data/order/OrderRepositoryTest.java]: can't you verify the actual input? making sure they match the expected ones?

File: core/src/main/java/com/kin/ecosystem/core/data/order/OrderRemoteData.java (hunk 1 above)
Comment (OrenZak) [reply core/src/main/java/com/kin/ecosystem/core/data/order/OrderRemoteData.java]: no need, generated swagger

File: core/src/test/java/com/kin/ecosystem/core/data/order/OrderRepositoryTest.java (hunk 3 above)
Comment (OrenZak) [reply core/src/test/java/com/kin/ecosystem/core/data/order/OrderRepositoryTest.java]: nope, the `Body` and `Callback` created inside the repo.

File: core/src/main/java/com/kin/ecosystem/core/network/model/Body.java (hunk 2 above)
Comment (OrenZak) [reply core/src/main/java/com/kin/ecosystem/core/network/model/Body.java]: We can use, but for future maybe we will want to add something extra without affecting the Error class. Also this is something generated from swagger and i don't wanna change that

**Pull request merged**